src/app/api/signup/route.ts

```diff
@@ -39,7 +39,18 @@ export const POST = withAppRouterHighlight(async (request: NextRequest) => {
     const userId = generateDatabaseId("user");
     const hashedPassword = await new Argon2id().hash(password);
 
-    // TODO: check if the user already exists
+    const userExists = await db
+      .selectFrom("user")
+      .selectAll()
+      .where("email", "=", parsedEmail.data)
+      .executeTakeFirst();
+
+    if (userExists) {
```

Comment:
Same reason as above, there are 2 ways to handle this.

1. Just log the user in
2. Send user an email saying this account already exists

Make sure the app should behave the same as if it is a normal signup, make a malicious actor not be able to tell apart if this user already has an account or not